src/java.base/share/classes/java/lang/System.java

```diff
@@ -2244,6 +2244,9 @@ private static void initPhase3() {
         // SecurityManager
         Unsafe.getUnsafe().ensureClassInitialized(StringConcatFactory.class);
 
+        // print out warning message if java.io.tmpdir is set through command line with non-existing folder
```

Comment:
It's probably okay to have the warning emitted in checkIoTmpDir, just a bit out of sight and inconsistent with the other warnings printed in this class.

I think the comment could be improved by bit with "Emit a warning if java.io.tmpdir is set via the command line to a directory that doesn't exist".